avm/res/data-protection/backup-vault/main.bicep

```diff
@@ -10,16 +10,41 @@ param enableTelemetry bool = true
 @description('Optional. Location for all resources.')
 param location string = resourceGroup().location
 
+import { roleAssignmentType } from 'br/public:avm/utl/types/avm-common-types:0.5.1'
 @description('Optional. Array of role assignments to create.')
-param roleAssignments roleAssignmentType
+param roleAssignments roleAssignmentType[]?
 
+import { lockType } from 'br/public:avm/utl/types/avm-common-types:0.5.1'
 @description('Optional. The lock settings of the service.')
-param lock lockType
+param lock lockType?
 
+import { managedIdentityAllType } from 'br/public:avm/utl/types/avm-common-types:0.5.1'
 @description('Optional. The managed identity definition for this resource.')
-param managedIdentities managedIdentitiesType
+param managedIdentities managedIdentityAllType?
```

Comment:
Please note, the the previous UDT supported **only** system-assigned

Reply:
The updated API supports both. Required for CMK with uami (preview). systemassigned+userassigned identity is also tested in both waf and cmk e2e tests